legend-engine-xt-sql-postgres-server/src/main/java/org/finos/legend/engine/postgres/handler/legend/LegendPreparedStatement.java

```diff
@@ -44,7 +50,20 @@ public void setObject(int i, Object o) throws Exception
     @Override
     public PostgresResultSetMetaData getMetaData() throws Exception
     {
-        return new LegendResultSetMetaData(client.getSchema(query));
+        if (identity.getFirstCredential() instanceof LegendKerberosCredential)
+
+        {
+            LegendKerberosCredential credential = (LegendKerberosCredential) identity.getFirstCredential();
+            return Subject.doAs(credential.getSubject(), (PrivilegedAction<LegendResultSetMetaData>) () ->
+            {
+                return new LegendResultSetMetaData(client.getSchema(query));
```

Comment:
Code in the `doAs` block and the else block look like duplicates. Can we extract that lambda to a private function and reuse it?